File: packages/amplify_core/lib/src/types/auth/attribute/auth_update_attribute_step.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import 'package:amplify_core/src/types/auth/attribute/auth_next_update_attribute_step.dart';

/// {@template amplify_core.auth.auth_update_attribute_step}
/// The current step in the attribute update flow.
/// {@endtemplate}
enum AuthUpdateAttributeStep {
  /// The next step is to confirm the attribute update with a code.
  ///
  /// Invoke `Amplify.Auth.confirmUserAttribute` with the attribute's key and
  /// the confirmation code delivered to the user specified via the
  /// [AuthNextUpdateAttributeStep.codeDeliveryDetails] field.
  confirmAttributeWithCode,

  /// The attribute update is complete and there are no remaining steps.
  done,
}

File: packages/amplify_datastore/darwin/amplify_datastore/Sources/amplify_datastore/AmplifyInternal/Categories/Auth/Models/AuthUpdateAttributeStep.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

/// Step for Auth.updateUserAttribute api call
public enum AuthUpdateAttributeStep {

    /// Next step is to confirm the attribute with confirmation code.
    ///
    /// Invoke Auth.confirm(userAttribute: ...) to confirm the attribute that was updated.
    /// `AuthCodeDeliveryDetails` provides the details to which the confirmation
    /// code was send and `AdditionalInfo` will provide more details if present.
    case confirmAttributeWithCode(AuthCodeDeliveryDetails, AdditionalInfo?)

    /// Update Attribute step is `done` when the update attribute flow is complete.
    case done
}
